plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/AbstractErrorMarkerResolution.java

```diff
@@ -61,4 +70,20 @@ public IMarker[] findOtherMarkers(final IMarker[] markers) {
 	public TypeLibrary getTypeLibrary() {
 		return TypeLibraryManager.INSTANCE.getTypeLibrary(marker.getResource().getProject());
 	}
+
+	protected static EObject getTargetElement(final IMarker marker) {
+		final URI uri = FordiacErrorMarker.getTargetUri(marker);
+		final TypeEntry typeEntry = TypeLibraryManager.INSTANCE.getTypeEntryForURI(uri);
+
+		final IEditorPart editor = Display.getDefault().syncCall(() -> PlatformUI.getWorkbench()
+				.getActiveWorkbenchWindow().getActivePage().findEditor(new FileEditorInput(typeEntry.getFile())));
+
+		LibraryElement libElement;
+		if (editor != null) {
+			libElement = Adapters.adapt(editor, LibraryElement.class);
+		} else {
+			libElement = typeEntry.getType();
```

Comment:
instead of getType you should use ` typeEntry.copyType()` as the `AbstractCommandChange` is doing it. Otherwise you may modify other users data. Am I right @mx990?

Reply:
Yes, please always either use the type from the editor or a copy created with `copyType()` when intending to  make changes. Never modify the type returned from `getType()` directly.

Reply:
Has been changed